packages/host/app/components/operator-mode/stack-item.gts

```diff
@@ -320,12 +321,15 @@ export default class OperatorModeStackItem extends Component<Signature> {
   };
 
   private initiateAutoSaveTask = restartableTask(async () => {
+    this.hasUnsavedChanges = true;
     await timeout(this.environmentService.autoSaveDelayMs);
     this.isSaving = true;
-    this.args.publicAPI.saveCard(this.card, false);
-    this.isSaving = false;
-    this.lastSaved = Date.now();
-    this.calculateLastSavedMsg();
+    this.args.publicAPI.saveCard(this.card, false, () => {
+      this.hasUnsavedChanges = false;
+      this.isSaving = false;
+      this.lastSaved = Date.now();
+      this.calculateLastSavedMsg();
+    });
```

Comment:
Why introduce an after callback instead of making saveCard async?

Reply:
good point. i was trying to be as "light touch" as possible with the API. but really there is no need to not be async

Reply:
actually wait-- there is a reason. I'm intentionally not awaiting an EC perform. When you do await an EC perform that actually casts it into an uncancellable promise. You should never `await` an EC perform outside of a task (inside of a task an `await` is actually syntactic sugar for a `yield`). more info about this is here: https://ember-concurrency.com/docs/task-cancelation-help